dask/dataframe/shuffle.py

```diff
@@ -186,6 +189,10 @@ def sort_values(
         # divisions are in the right place
         return df.map_partitions(sort_function, **sort_kwargs)
 
+    # sometimes `_calculate_divisions`` will give us divisions containing pd.NA,
+    # which can't be used in `searchsorted` as part of `rearrange_by_divisions`
+    divisions = [np.NaN if x is pd.NA else x for x in divisions]
+
```

Comment:
Can we move this to be closer to where it's needed? It looks like this is helping make some `searchsorted` calls work -- could we move this logic closer to those calls?